### PR TITLE
shell auto-configuration

### PR DESCRIPTION
**Pull request by joshlong**
this contributes `@EnableDataFlowShell` and required configuration.

### CODE REVIEW

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/EnableDataFlowShell.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.springframework.cloud.dataflow.shell;
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/EnableDataFlowShell.java]: missing copyright header

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BootstrapConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
+package org.springframework.cloud.dataflow.shell.autoconfigure;
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BootstrapConfiguration.java]: missing copyright header

**Comment by markfisher** [comment]
The `spring-cloud-dataflow` build fails on this branch. Both of these test classes within the shell sub-project are failing: `StreamCommandTests` and `TaskCommandTests`

**Comment by joshlong** [comment]
thanks @markfisher for taking a look! 

i just added the two copyright headers but im confused what you mean the build was failing. Travis was green and my local build was green (mvn -f  spring-cloud-dataflow/pom.xml  clean install ) works..

**Comment by jvalkeal** [comment]
Tests ran fine when I tried.

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/EnableDataFlowShell.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015 the original author or authors.
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/EnableDataFlowShell.java]: it's 2016

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BaseShellAutoConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 the original author or authors.
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BaseShellAutoConfiguration.java]: 2016

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BaseShellAutoConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.shell.autoconfigure;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BaseShellAutoConfiguration.java]: should be an empty line before `org.springframework.*` imports

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BootstrapConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2015 the original author or authors.
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BootstrapConfiguration.java]: 2016

**Comment by markfisher** [comment]
`spring-shell-plugin.xml` was moved under `src/test`... is it even needed anymore?

**Comment by joshlong** [comment]
@markfisher  its' needed because the two tests that use the shell still use the old style `Bootstrap` class since creating a shell and a server in the same VM and not having auto configuration step on each other was doubly painful.

**Comment by joshlong** [comment]
tests pass on my machine, updated headers, added space between `o.s.*` imports, and even remembered to add `@EnableDataFlowShell` to the `ShellApplication` so that `java -jar target/spring-cloud-dataflow-shell-1.0.0.BUILD-SNAPSHOT.jar` works as expected :D HURRAY

**Comment by markfisher** [comment]
@ericbottard can you please take a quick look at this PR? If you're okay with it, you could merge it. Otherwise assuming it looks good to you, I will merge first thing in the morning.

**Comment by ericbottard** [comment]
Will do

File: spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java [hunk 7]
```diff
@@ -109,19 +110,23 @@ public static void startUp() throws InterruptedException, IOException {
...
+			SpringApplication application = new SpringApplicationBuilder(TestConfig.class).build();
+
 			applicationContext = application.run(
-					String.format("--server.port=%s", serverPort), "--security.basic.enabled=false",
+					String.format("--server.port=%s", serverPort),
+					"--spring.jmx.default-domain=" + System.currentTimeMillis(),
+					"--spring.jmx.enabled=false",
+					"--security.basic.enabled=false",
 					"--spring.main.show_banner=false", "--spring.cloud.config.enabled=false",
 					"--deployer.local.out-of-process=false");
```
Comment (ericbottard) [inline spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java]: This should be removed I believe

**Comment by ericbottard** [comment]
Looks good to me overall

**Comment by ericbottard** [comment]
Squashed and merged as 95c5fb1, including the removal of the property I mentioned above.

**Comment by markfisher** [comment]
Thanks @ericbottard ! and thanks @joshlong for contributing this.... nice improvement!

**Pull request closed without merging**